src/Text/Pandoc/Readers/RST.hs

```diff
@@ -775,17 +775,18 @@ tableDirective top _fields body = do
 listTableDirective :: PandocMonad m => String -> [(String, String)] -> String -> RSTParser m Blocks
 listTableDirective top _fields body = do
   bs <- parseFromString parseBlocks body
-  title <- do { parseFromString (trimInlines . mconcat <$> many inline) top }
+  title <- parseFromString (trimInlines . mconcat <$> many inline) top
   let rows = takeRows $ B.toList bs
-      numOfCols = length $ rows !! 0
-      headerRows = case lookup "header-rows" _fields of
-                     Just x -> read x :: Int
-                     Nothing -> 0
+      headerRow = case rows of
```

Comment:
I'd suggest something like
```
(firstRow, restRows) = case rows of
                 (x:xs) -> (x, xs)
                 [] -> ([], [])
```
Then you don't have to do the `drop 1 rows` below.  (This should be a *tiny* bit more efficient.)